test/plugins/compliance_visitor_data_request_plugin_test.rb

```diff
@@ -0,0 +1,159 @@
+        ApiResource.create(api_namespace_id: @namespace_with_email_one.id, properties: {"post": "Hello world", "email": @visitor_one.properties["email"]})
+        ApiResource.create(api_namespace_id: @namespace_with_email_one.id, properties: {"post": "Hello world", "email": @visitor_two.properties["email"]})
+        assert_difference "ActionMailer::Base.deliveries.size", +2 do
+            perform_enqueued_jobs do
+                @data_request_plugin.run
+                Sidekiq::Worker.drain_all
+            end 
+        end
+    end
+
+    test "Sender and recipient email addresses and body message are correct" do
+        visitor_email = @visitor_one.properties["email"]
+        ApiResource.create(api_namespace_id: @namespace_with_email_one.id, properties: {"post": "Hello world", "email": visitor_email})
+
+        expected_message = @data_request_plugin.metadata["MESSAGE"]
+
+        perform_enqueued_jobs do
+            @data_request_plugin.run
+            Sidekiq::Worker.drain_all
+        end
+
+        sent_email = ActionMailer::Base.deliveries.last
+        assert(sent_email.from.include?(Subdomain.current.name))
+        assert(sent_email.from.include?(ENV["APP_HOST"]))
+        assert_equal [visitor_email], sent_email.to
+        assert(sent_email.parts.first.body.raw_source.include?(expected_message))
+    end
+
+    test "Should not scan excluded api namespaces" do
+        @data_request_plugin.metadata["EXCLUDE_API_NAMESPACES"] << @namespace_with_email_one.slug
+        @data_request_plugin.save
+        visitor_email = @visitor_one.properties["email"]
+        ApiResource.create(api_namespace_id: @namespace_with_email_one.id, properties: {"post": "Hello world", "email": visitor_email})
+        # Should not send an email even if a submission was made under a namespace since it's excluded
+        assert_no_difference "ActionMailer::Base.deliveries.size" do
+            perform_enqueued_jobs do
+                @data_request_plugin.run
+                Sidekiq::Worker.drain_all
+            end
+        end
+    end
+
+    test "Email should contain the right number of CSV attachments" do
+        visitor_email = @visitor_one.properties["email"]
+        ApiResource.create(api_namespace_id: @namespace_with_email_one.id, properties: {"post": "Hello world", "email": visitor_email})
+        ApiResource.create(api_namespace_id: @namespace_with_email_two.id, properties: {"name": "Test Name", "email": visitor_email})
+
+        perform_enqueued_jobs do
+            @data_request_plugin.run
+            Sidekiq::Worker.drain_all
+        end
+
+        # Visitor one making submissions under two namespaces, so the email should contain two attachments
+        sent_email = ActionMailer::Base.deliveries.last
+        assert_equal 2, sent_email.attachments.length
+
+        sent_email.attachments.each do |attachment|
+            assert(attachment.content_type.start_with?("text/csv"))
```

Comment:
can you also assert that the two CSV's are different (since they are pertaining to a different API Namespace) and include the data for its respective API Namespace (eg: one API namespaces has attributes: `name`,` email`, `phone` -- while the other has `address` and `patient_id`

Reply:
Like this?
```ruby
assert_not(first_attachment_content == second_attachment_content)

@namespace_with_email_one.properties.each do |key, value|
   assert(first_attachment_content.include?(key))
   assert(first_attachment_content.include?(value))
end

@namespace_with_email_two.properties.each do |key, value|
  assert(second_attachment_content.include?(key))
  assert(second_attachment_content.include?(value))
end
```

Reply:
@Ayon95 yup like that